job-service/src/main/java/com/hpe/caf/services/job/api/JobsPut.java

```diff
@@ -188,46 +177,28 @@ public static String createOrUpdateJob(final String partitionId, String jobId, N
                     partitionSuspended ? "suspended partition" : "partition", partitionId);
             final boolean jobCreated;
             if ((job.getPrerequisiteJobIds() != null && !job.getPrerequisiteJobIds().isEmpty()) || partitionSuspended) {
+                hasDependencies = true;
                 jobCreated = databaseHelper.createJobWithDependencies(partitionId, jobId, job.getName(), job.getDescription(),
                         job.getExternalData(), jobHash, jobTask.getTaskClassifier(), jobTask.getTaskApiVersion(),
                         getTaskDataBytes(jobTask, codec), jobTask.getTaskPipe(), jobTask.getTargetPipe(),
                         job.getPrerequisiteJobIds(), job.getDelay(), job.getLabels(), partitionSuspended);
 
             } else {
-                jobCreated = databaseHelper.createJob(partitionId, jobId, job.getName(), job.getDescription(), job.getExternalData(), jobHash, job.getLabels());
+                hasDependencies = false;
+                jobCreated = databaseHelper.createJob(partitionId, jobId, job.getName(), job.getDescription(),
+                        job.getExternalData(), jobHash, jobTask.getTaskClassifier(), jobTask.getTaskApiVersion(),
+                        getTaskDataBytes(jobTask, codec), jobTask.getTaskPipe(), jobTask.getTargetPipe(),
+                        job.getDelay(), job.getLabels());
             }
 
             if (!jobCreated) {
                 return "update";
             }
 
-            if (!databaseHelper.canJobBeProgressed(partitionId, jobId)) {
+            if (hasDependencies && !databaseHelper.canJobBeProgressed(partitionId, jobId)) {
```

Comment:
What is the purpose of this?  It looks like you are returning `"create"` either way.

Reply:
True, strangely we were doing it before and I just tried to adapt. I guess there was a use for it some time ago, but you're right, not anymore and it can be removed. Also, that's the only place in the code where we use canJobBeProgressed() so I cleaned that one too